### PR TITLE
CORL-2358 gracefully close connections

### PR DESCRIPTION
**Pull request by marcushaddon**
## What does this PR do?
This PR addresses https://vmproduct.atlassian.net/browse/CORL-2348 by closing connections on the express server on SIGTERM, as well as setting up a method in which any other resources can be cleaned up if need be.

## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## How do I test this PR?
TODO
 
## How do we deploy this PR?
No special considerations should be required.

### CODE REVIEW

File: src/core/server/index.ts [hunk 1]
```diff
@@ -278,6 +278,16 @@ class Server {
...
   }
 
+  /**
+   * stop gracefully closes connections and cleans up any resources.
+   */
+  public async stop() {
+    if (this.httpServer) {
+      this.httpServer.close(() => logger.info("Http server closed"));
+    }
+    // Do we need additional cleanup here?
```
Comment (marcushaddon) [inline src/core/server/index.ts:288]: ie mongo/redis connections?

**Review by alinhle — CHANGES_REQUESTED** [review]
Screenshot 2021-11-10 at 14 34 25

Unfortunately only closing the `httpServer` is not enough, because all of our connections are long living (keep-alive) tcp connections. 

Try finding a library that helps closing these connections as well.

File: src/core/server/index.ts (hunk 1 above)
Comment (alinhle) [reply src/core/server/index.ts:288]: That won't be necessary.

**Comment by tessalt** [comment]
closing until we can re-evaluate options.

**Pull request closed without merging**